internal_ws/ykcompile/src/lib.rs

```diff
@@ -57,9 +57,17 @@ lazy_static! {
     static ref ICTX_REG: u8 = RDI.code();
 
     static ref TEMP_LOC: Location = Location::Reg(*TEMP_REG);
+
+    /// The size of a memory address in bytes.
     static ref PTR_SIZE: u64 = u64::try_from(mem::size_of::<usize>()).unwrap();
 }
 
+/// The size of a register in bytes.
+const REG_SIZE: usize = 8;
```

Comment:
Not all registers are the same size, so this name feels a bit confusing?

Reply:
we can prefix it `GP` for "general purpose".

Reply:
Maybe a less-worse way would be to have a `const HashMap` from register names to sizes, or simply one constant per register? Or is this a generic "make sure all registers can fit into this" constant or ... ?

Reply:
I'm feel uneasy about maintaining tables of registers, as I think it would make the code even harder to read than it already is, and (at least for now) there's no real need to add indirection.

In general CPUs tend to have "classes" of registers, where each register in a class is the same size. But as usual, it's not that simple. In x86_64, general purpose registers are partitioned into 1-, 2- , 4- and 8-byte registers. It's all an upshot of legacy.

In the above case, a better name for my constant would be `QWORD_REG_SIZE`, where `Q` stands for quad-word (because in x86_64, strictly a word is 16-bytes).

Reply:
QWORD_REG_SIZE seems reasonable.

Reply:
Fixed in e278c3a